webapp/src/main/java/org/opentestsystem/rdw/admin/repository/impl/JdbcGroupRepository.java

```diff
@@ -36,20 +44,47 @@
                 new MapSqlParameterSource()
                         .addValues(getSecurityParameters(permissionScope))
                         .addValues(getQueryParameters(query)),
-                (row, index) -> Group.builder()
-                        .name(row.getString("group_name"))
-                        .schoolName(row.getString("school_name"))
-                        .schoolYear(row.getInt("school_year"))
-                        .subject(row.getString("subject_code"))
-                        .build()
+                (row, index) -> mapResultToGroup(row)
+        );
+    }
+
+    @Override
+    public List<Integer> findAllDistinctSchoolYears(@NotNull final PermissionScope permissionScope) {
+        return template.query(
+                findAllDistinctSchoolYears,
+                getSecurityParameters(permissionScope),
+                (row, index) -> row.getInt("school_year")
         );
     }
 
+    private static Group mapResultToGroup(final ResultSet row) throws SQLException {
+        return Group.builder()
+                .name(row.getString("group_name"))
+                .schoolName(row.getString("school_name"))
+                .schoolYear(row.getInt("school_year"))
+                .subject(parseSubject(row.getString("subject_id")))
+                .build();
+    }
+
     private static Map<String, Object> getQueryParameters(final GroupQuery query) {
         return ImmutableMap.<String, Object>builder()
                 .put("school_id", query.getSchoolId())
-                .put("subject_ids", query.getSubjectIds())
+                .put("subject_ids", query
+                        .getSubjects()
+                        .stream()
+                        .map(Subject::id)
+                        .collect(Collectors.toSet()))
```

Comment:
I hate the .stream() and .collect() Java makes you do when all you want to do is a simple mapping.  Is there an easier way?